manifests/container.pp

```diff
@@ -0,0 +1,49 @@
+# @summary Pulp Container plugin
+# @param location_prefix
+#   In the Apache configuration a location with this prefix is exposed. The
+#   version will be appended.
+# @param registry_v1_path
+#   The path beneath the location prefix to forward. This is also appended to
+#   the content base url.
+# @param registry_v2_path
+#   The path beneath the location prefix to forward. This is also appended to
+#   the content base url.
+class foreman_proxy_content::container (
+  String $location_prefix = '/pulpcore_registry',
+  String $registry_v1_path = '/v1/',
+  String $registry_v2_path = '/v2/',
+) {
+
+  $context = {
+    'directories'  => [
+      {
+        'provider'        => 'location',
+        'path'            => "${location_prefix}${$registry_v2_path}",
+        'request_headers' => ["set SSL_CLIENT_S_DN \"admin\""],
+        'requires'        => ["expr %{SSL_CLIENT_S_DN_CN} == \"${facts['networking']['fqdn']}\""]
```

Comment:
This shouldn't be the FQDN but rather the hostname on the client cert right? So I'd `include certs::foreman_proxy` and then use `$certs::foreman_proxy::hostname`. Not sure if the exact variables are correct.

The short summary is: use variables from classes that actually ensure the resources.